### ISSUE TITLE
carpet-fixes conflicts with Immersive Portals Mod

### ISSUE THREAD

**Issue by misaka4a92** [fxmorin/carpet-fixes#38]
Here is some of my environment:

- minecraft 1.18.1
- Fabric Loader 0.13.3
- fabric-api-0.46.6+1.18
- fabric-carpet-1.18.1-1.4.57+v220119
- carpet-fixes-1.18.x-1.7.1+v220311
- immersive-portals-1.2.6-mc1.18.1-fabric

I believe that the mixins <https://github.com/qouteall/ImmersivePortalsMod/blob/59bd88a97e8e9810ae05d2f7a5ee3fd88202bc85/imm_ptl_core/src/main/java/qouteall/imm_ptl/core/mixin/common/MixinServerLevel.java#L39> and <https://github.com/fxmorin/carpet-fixes/blob/038a711afff78e5a856d86b98ba01d85af4277e9/src/main/java/carpetfixes/mixins/coreSystemFixes/ServerWorld_spawnChunksMixin.java#L21> conflicts. 

I hope this issues can be fixed for 1.18.1 and other versions. Thanks.

and related log message:
```
[24:60:60] [main/WARN]: @Redirect conflict. Skipping imm_ptl.mixins.json:common.MixinServerLevel->@Redirect::redirectIsEmpty(Ljava/util/List;)Z from mod imm_ptl_core with priority 1000, already redirected by carpet-fixes.mixins.json:coreSystemFixes.ServerWorld_spawnChunksMixin->@Redirect::spawnChunksStayLoaded(Ljava/util/List;)Z from mod carpet-fixes with priority 1000
...
[24:60:60] [main/ERROR]: Minecraft has crashed!
net.fabricmc.loader.impl.FormattedException: java.lang.ExceptionInInitializerError
	at Not Enough Crashes deobfuscated stack trace.(1.18.1+build.22) ~[?:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:418) ~[fabric-loader-0.13.3.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:77) [fabric-loader-0.13.3.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) [fabric-loader-0.13.3.jar:?]
Caused by: java.lang.ExceptionInInitializerError
	at net.minecraft.world.gen.carver.CarverDebugConfig.<clinit>(CarverDebugConfig:9) ~[?:?]
	at net.minecraft.world.gen.carver.CarverConfig.method_33968(CarverConfig:16) ~[?:?]
	at com.mojang.serialization.codecs.RecordCodecBuilder.mapCodec(RecordCodecBuilder.java:76) ~[datafixerupper-4.0.26.jar:?]
	at net.minecraft.world.gen.carver.CarverConfig.<clinit>(CarverConfig:12) ~[?:?]
	at net.minecraft.world.gen.carver.Carver.<clinit>(Carver:31) ~[?:?]
	at net.minecraft.world.gen.carver.ConfiguredCarvers.<clinit>(ConfiguredCarvers:18) ~[?:?]
	at net.minecraft.util.registry.BuiltinRegistries.method_30572(BuiltinRegistries:46) ~[?:?]
	at net.minecraft.util.registry.BuiltinRegistries.method_30566(BuiltinRegistries:78) ~[?:?]
	at java.util.LinkedHashMap.forEach(Unknown Source) ~[?:?]
	at net.minecraft.util.registry.BuiltinRegistries.<clinit>(BuiltinRegistries:77) ~[?:?]
	at net.minecraft.util.registry.Registry.<clinit>(Registry:269) ~[?:?]
	at net.minecraft.Bootstrap.initialize(Bootstrap:49) ~[?:?]
	at net.minecraft.client.main.Main.main(Main:155) ~[1.18.1.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:?]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:416) ~[fabric-loader-0.13.3.jar:?]
	... 2 more
Caused by: java.lang.RuntimeException: Mixin transformation of net.minecraft.class_3218 failed
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:252) ~[fabric-loader-0.13.3.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:150) ~[fabric-loader-0.13.3.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:155) ~[fabric-loader-0.13.3.jar:?]
	at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:?]
	at net.minecraft.block.Blocks.<clinit>(Blocks:103) ~[?:?]
	at net.minecraft.world.gen.carver.CarverDebugConfig.<clinit>(CarverDebugConfig:9) ~[?:?]
	at net.minecraft.world.gen.carver.CarverConfig.method_33968(CarverConfig:16) ~[?:?]
	at com.mojang.serialization.codecs.RecordCodecBuilder.mapCodec(RecordCodecBuilder.java:76) ~[datafixerupper-4.0.26.jar:?]
	at net.minecraft.world.gen.carver.CarverConfig.<clinit>(CarverConfig:12) ~[?:?]
	at net.minecraft.world.gen.carver.Carver.<clinit>(Carver:31) ~[?:?]
	at net.minecraft.world.gen.carver.ConfiguredCarvers.<clinit>(ConfiguredCarvers:18) ~[?:?]
	at net.minecraft.util.registry.BuiltinRegistries.method_30572(BuiltinRegistries:46) ~[?:?]
	at net.minecraft.util.registry.BuiltinRegistries.method_30566(BuiltinRegistries:78) ~[?:?]
	at java.util.LinkedHashMap.forEach(Unknown Source) ~[?:?]
	at net.minecraft.util.registry.BuiltinRegistries.<clinit>(BuiltinRegistries:77) ~[?:?]
	at net.minecraft.util.registry.Registry.<clinit>(Registry:269) ~[?:?]
	at net.minecraft.Bootstrap.initialize(Bootstrap:49) ~[?:?]
	at net.minecraft.client.main.Main.main(Main:155) ~[1.18.1.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:?]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:416) ~[fabric-loader-0.13.3.jar:?]
	... 2 more
Caused by: org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:392) ~[sponge-mixin-0.11.2+mixin.0.8.5.jar:0.11.2+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234) ~[sponge-mixin-0.11.2+mixin.0.8.5.jar:0.11.2+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202) ~[sponge-mixin-0.11.2+mixin.0.8.5.jar:0.11.2+mixin.0.8.5]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:247) ~[fabric-loader-0.13.3.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:150) ~[fabric-loader-0.13.3.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:155) ~[fabric-loader-0.13.3.jar:?]
	at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:?]
	at net.minecraft.block.Blocks.<clinit>(Blocks:103) ~[?:?]
	at net.minecraft.world.gen.carver.CarverDebugConfig.<clinit>(CarverDebugConfig:9) ~[?:?]
	at net.minecraft.world.gen.carver.CarverConfig.method_33968(CarverConfig:16) ~[?:?]
	at com.mojang.serialization.codecs.RecordCodecBuilder.mapCodec(RecordCodecBuilder.java:76) ~[datafixerupper-4.0.26.jar:?]
	at net.minecraft.world.gen.carver.CarverConfig.<clinit>(CarverConfig:12) ~[?:?]
	at net.minecraft.world.gen.carver.Carver.<clinit>(Carver:31) ~[?:?]
	at net.minecraft.world.gen.carver.ConfiguredCarvers.<clinit>(ConfiguredCarvers:18) ~[?:?]
	at net.minecraft.util.registry.BuiltinRegistries.method_30572(BuiltinRegistries:46) ~[?:?]
	at net.minecraft.util.registry.BuiltinRegistries.method_30566(BuiltinRegistries:78) ~[?:?]
	at java.util.LinkedHashMap.forEach(Unknown Source) ~[?:?]
	at net.minecraft.util.registry.BuiltinRegistries.<clinit>(BuiltinRegistries:77) ~[?:?]
	at net.minecraft.util.registry.Registry.<clinit>(Registry:269) ~[?:?]
	at net.minecraft.Bootstrap.initialize(Bootstrap:49) ~[?:?]
	at net.minecraft.client.main.Main.main(Main:155) ~[1.18.1.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:?]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:416) ~[fabric-loader-0.13.3.jar:?]
	... 2 more
Caused by: org.spongepowered.asm.mixin.injection.throwables.InjectionError: Critical injection failure: Redirector redirectIsEmpty(Ljava/util/List;)Z in imm_ptl.mixins.json:common.MixinServerLevel from mod imm_ptl_core failed injection check, (0/1) succeeded. Scanned 1 target(s). Using refmap imm_ptl_core-refmap.json
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.postInject(InjectionInfo.java:468) ~[sponge-mixin-0.11.2+mixin.0.8.5.jar:0.11.2+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTargetContext.applyInjections(MixinTargetContext.java:1385) ~[sponge-mixin-0.11.2+mixin.0.8.5.jar:0.11.2+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyInjections(MixinApplicatorStandard.java:1052) ~[sponge-mixin-0.11.2+mixin.0.8.5.jar:0.11.2+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyMixin(MixinApplicatorStandard.java:400) ~[sponge-mixin-0.11.2+mixin.0.8.5.jar:0.11.2+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:325) ~[sponge-mixin-0.11.2+mixin.0.8.5.jar:0.11.2+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.apply(TargetClassContext.java:421) ~[sponge-mixin-0.11.2+mixin.0.8.5.jar:0.11.2+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:403) ~[sponge-mixin-0.11.2+mixin.0.8.5.jar:0.11.2+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363) ~[sponge-mixin-0.11.2+mixin.0.8.5.jar:0.11.2+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234) ~[sponge-mixin-0.11.2+mixin.0.8.5.jar:0.11.2+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202) ~[sponge-mixin-0.11.2+mixin.0.8.5.jar:0.11.2+mixin.0.8.5]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:247) ~[fabric-loader-0.13.3.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:150) ~[fabric-loader-0.13.3.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:155) ~[fabric-loader-0.13.3.jar:?]
	at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:?]
	at net.minecraft.block.Blocks.<clinit>(Blocks:103) ~[?:?]
	at net.minecraft.world.gen.carver.CarverDebugConfig.<clinit>(CarverDebugConfig:9) ~[?:?]
	at net.minecraft.world.gen.carver.CarverConfig.method_33968(CarverConfig:16) ~[?:?]
	at com.mojang.serialization.codecs.RecordCodecBuilder.mapCodec(RecordCodecBuilder.java:76) ~[datafixerupper-4.0.26.jar:?]
	at net.minecraft.world.gen.carver.CarverConfig.<clinit>(CarverConfig:12) ~[?:?]
	at net.minecraft.world.gen.carver.Carver.<clinit>(Carver:31) ~[?:?]
	at net.minecraft.world.gen.carver.ConfiguredCarvers.<clinit>(ConfiguredCarvers:18) ~[?:?]
	at net.minecraft.util.registry.BuiltinRegistries.method_30572(BuiltinRegistries:46) ~[?:?]
	at net.minecraft.util.registry.BuiltinRegistries.method_30566(BuiltinRegistries:78) ~[?:?]
	at java.util.LinkedHashMap.forEach(Unknown Source) ~[?:?]
	at net.minecraft.util.registry.BuiltinRegistries.<clinit>(BuiltinRegistries:77) ~[?:?]
	at net.minecraft.util.registry.Registry.<clinit>(Registry:269) ~[?:?]
	at net.minecraft.Bootstrap.initialize(Bootstrap:49) ~[?:?]
	at net.minecraft.client.main.Main.main(Main:155) ~[1.18.1.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:?]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:416) ~[fabric-loader-0.13.3.jar:?]
	... 2 more
```

**Comment by FxMorin** [comment]
Oh this looks like a fun one

**Comment by FxMorin** [comment]
Can you please check with the latest release and tell me if you still get the issue, thanks!

**Comment by misaka4a92** [comment]
Thanks for your quick fix, release 1.7.2 works perfectly except a log message ``[main/WARN]: Mod `carpet-fixes` (1.18.x-1.7.2+v220318) does not respect SemVer - comparison support is limited.``. There is no noticable issues now.
